controllers/controller_shared.go

```diff
@@ -87,9 +87,15 @@ func GetScopedMatchingInstances(ctx context.Context, k8sClient client.Client, cr
 		if !selected {
 			continue
 		}
+
+		// Readiness checks omits instances that have not completed a full reconciliation successfully.
+		// This allows speeding up reconciliations and reduces the amount of noisy errors.
+		// A toggle to skip the readiness check allows additional testing of reconcilers, like provoking the ApplyFailed synchronization condition.
+		doReadinessCheck := instance.Annotations["grafana/skip-readiness-check"] != "true"
```

Comment:
As mentioned in the original PR, the annotation should be named after the operator, not after grafana to avoid confusion.